crates/ruff_workspace/src/configuration.rs

```diff
@@ -322,26 +296,14 @@ impl Configuration {
     }
 
     pub fn from_options(options: Options, project_root: &Path) -> Result<Self> {
+        // TODO warn about legacy options
+        let lint = if let Some(lint) = options.lint {
+            lint.combine(options.lint_top_level)
+        } else {
+            options.lint_top_level
+        };
```

Comment:
Having the options vs. configuration separation is useful here!

Reply:
It is. It also makes sense in many other places. I'm fairly happy with our `Options` -> `Configuration` -> `Settings` separation